README.md

```diff
@@ -37,63 +32,237 @@ Let's look at each of the components in more detail.
 ### Schema
 
 A model of a schema with helpful methods for working with the schema and supported data. Schema instances can be initialized with a schema source as a url to a JSON file or a JSON object. 
-The schema is initially validated (see [validate](#validate) below), and will raise an exception if not a valid JSON Table Schema.
+The schema is initially validated (see [validate](#validate) below), and will raise an exception if not a valid JSON Table Schema.  
 
-Model requires Promise to work properly, and need to be sure that Promise available globally. You are free to choose any Promise polyfill  
+```javascript
+var Schema = require('jsontableschema').schema;
+```
 
 ```javascript
-const model = new Schema('http://someurl.com/remote.json')
+var model = new Schema('http://someurl.com/remote.json')
 ```
 or
 ```javascript
-const model = new Schema({JSON OBJECT})
+var model = new Schema({JSON OBJECT})
 ```
-instance always returns promise
+instance always returns `Promise`
 ```javascript
-model.then(schema => {
+model.then(function(schema) {
     // working code to use schema model
-}).catch(error => {
+    var fields = schema.fields();
+   
+}).catch(function(error) {
     // something went wrong and error variable has explanations
 })
 ```
 
-Some methods available to Schema instances:
+Following methods are available on `Schema` instances:
 
-* `cast(fieldName, value, index, skipConstraints)` - returns a value cast against a named `fieldName`
-* `convertRow(...args)` - convert the arguments given to the types of the current schema <sup>1</sup>
-* `convert(items, failFast = false)` - convert an array of rows using the current schema of the Schema instance <sup>2</sup>
+* `castValue(fieldName, value, index, skipConstraints)` - returns a value cast against a named `fieldName`
+* `testValue(fieldName, value, index, skipConstraints)` - returns boolean after a check if value can be casted against a named `fieldName`
+* `castRow(items, failFast = false, skipConstraints = false)` - convert the arguments given to the types of the current schema <sup>1</sup>
 * `fields` - returns an array of the schema's fields
 * `foreignKeys` - returns the foreign key property for the schema
```

Comment:
it's marked in jsdoc as array but we could highlight it here also (it's array of objects)